go/integTest/cluster_commands_test.go

```diff
@@ -294,3 +295,448 @@ func (suite *GlideTestSuite) TestEchoCluster() {
+}
+
+func (suite *GlideTestSuite) TestBasicClusterScanWithOptions() {
+	client := suite.defaultClusterClient()
+	t := suite.T()
+
+	// Ensure clean start
+	_, err := client.CustomCommand([]string{"FLUSHALL"})
+	assert.NoError(t, err)
+
+	// Iterate over all keys in the cluster
+	keysToSet := map[string]string{
+		"key1": "value1",
+		"key2": "value2",
+		"key3": "value3",
+	}
+
+	_, err = client.MSet(keysToSet)
+	assert.NoError(t, err)
+
+	cursor := *options.NewClusterScanCursor()
+	opts := options.NewClusterScanOptions().SetCount(10)
+	allKeys := []string{}
+
+	for !cursor.HasFinished() {
+		nextCursor, keys, err := client.ScanWithOptions(&cursor, *opts)
+		if err != nil {
+			assert.NoError(t, err) // Use this to print error statement
+			break                  // prevent infinite loop
+		}
+		allKeys = append(allKeys, keys...)
+
+		cursor = *options.NewClusterScanCursorWithId(nextCursor)
+	}
+
+	assert.ElementsMatch(t, allKeys, []string{"key1", "key2", "key3"})
+
+	// Iterate over keys matching a pattern
+	keysToSet = map[string]string{
+		"key1":          "value1",
+		"key2":          "value2",
+		"notMykey":      "value3",
+		"somethingElse": "value4",
+	}
+
+	_, err = client.MSet(keysToSet)
+	assert.NoError(t, err)
+
+	cursor = *options.NewClusterScanCursor()
+	opts = options.NewClusterScanOptions().SetCount(10).SetMatch("*key*")
+	matchedKeys := []string{}
+
+	for !cursor.HasFinished() {
+		nextCursor, keys, err := client.ScanWithOptions(&cursor, *opts)
+		if err != nil {
+			assert.NoError(t, err) // Use this to print error statement
+			break                  // prevent infinite loop
+		}
+		matchedKeys = append(matchedKeys, keys...)
+
+		cursor = *options.NewClusterScanCursorWithId(nextCursor)
+	}
+
+	assert.ElementsMatch(t, matchedKeys, []string{"key1", "key2", "key3", "notMykey"})
+	assert.NotContains(t, matchedKeys, "somethingElse")
+
+	// Iterate over keys of a specific type
+	keysToSet = map[string]string{
+		"key1": "value1",
+		"key2": "value2",
+		"key3": "value3",
+	}
+	_, err = client.MSet(keysToSet)
+	assert.NoError(t, err)
+
+	_, err = client.SAdd("thisIsASet", []string{"someValue"})
+	assert.NoError(t, err)
+
+	cursor = *options.NewClusterScanCursor()
+	opts = options.NewClusterScanOptions().SetType(options.ObjectTypeSet)
+	matchedTypeKeys := []string{}
+
+	for !cursor.HasFinished() {
+		nextCursor, keys, err := client.ScanWithOptions(&cursor, *opts)
+		if err != nil {
+			assert.NoError(t, err) // Use this to print error statement
+			break                  // prevent infinite loop
+		}
+		matchedTypeKeys = append(matchedTypeKeys, keys...)
+
+		cursor = *options.NewClusterScanCursorWithId(nextCursor)
+	}
+
+	assert.ElementsMatch(t, matchedTypeKeys, []string{"thisIsASet"})
+	assert.NotContains(t, matchedTypeKeys, "key1")
+	assert.NotContains(t, matchedTypeKeys, "key2")
+	assert.NotContains(t, matchedTypeKeys, "key3")
+}
+
+func (suite *GlideTestSuite) TestBasicClusterScanWithInvalidUTF8Pattern() {
```

Comment:
I'd rename the test to "...NonUtf8Patter"
The word "invalid" triggered me to find an error case

Reply:
Thank you for testing that though